docs(task-delegation): document Task Delegation feature and release notes (#3367)
bonitasoft/bonita-doc: modules/ROOT/pages/release-notes.adoc

== New available values

=== Task Delegation

[NOTE]
====
For Teamwork, Efficiency, Performance, Enterprise and Scale editions only. +
====

> When a user is away, their pending tasks normally stay invisible to everyone else. Task Delegation lets a designated delegate see and perform those tasks on behalf of the absent user, without reassigning them, and keeps full traceability of who acted.

Reviewer: ```suggestion
When a user is away, their pending assigned tasks normally stay invisible to everyone else. Task Delegation lets a designated delegate see and perform those tasks on behalf of the absent user, without reassigning them, and keeps full traceability of who acted.
```

> This feature is delivered through a new Subscription living application, *Bonita User Application Enterprise*, which replaces the community *Bonita User Application* and adds the delegation pages.

Reviewer: ```suggestion
This feature is delivered through a new *Enterprise edition Bonita User Application*, which replaces the existing Bonita User Application and adds the delegation page.
```

> *On a fresh installation*, the *Bonita User Application Enterprise* is installed automatically. Nothing else to do.

Reviewer: ```suggestion
*On a fresh installation*, the Enterprise edition user application is installed automatically. Nothing else to do.
```

> *When updating from a previous Bonita version*, your existing *Bonita User Application* is kept and the new *Bonita User Application Enterprise* is added alongside it, so the two applications coexist. A manual step is required to finish the setup, depending on your situation:

Reviewer: ```suggestion
*When updating from a previous Bonita version*, your existing user application is kept and the new Enterprise edition user application is added alongside it, so the two applications coexist. A manual step is required to finish the setup, depending on your situation:
```

. *You did not customize the Bonita Administrator Application*: you have nothing to do.
. *You customized the Bonita Administrator Application*: in the application descriptor of your custom admin application, add a new Delegations menu entry under BPM targeting the admin Delegations page (`page-admin-delegation`).

[NOTE]
====
This release ships updated versions of the provided xref:identity:rest-api-authorization.adoc#dynamic_authorization[REST API permission rules]. The task endpoints' rules `TaskPermissionRule` and `TaskExecutionPermissionRule` are updated so that an active delegate is granted access to the delegated tasks, and a new `DelegationPermissionRule` secures the delegation endpoints (`delegation/rule` and `delegation/task`), with the matching entries added to `dynamic-permissions-checks.properties`.

If you customized any of the provided permission rules, you need to update your custom rules to re-apply your changes on top of the new versions of those rules after upgrading (otherwise the delegation feature will not work). See xref:identity:rest-api-authorization.adoc#dynamic_authorization[Dynamic authorization checking].
====

==== Known limitations

> Business data (BDM) access is controlled separately from delegation. If a task form reads or writes business data, the delegate must have the same BDM access rights as the delegator to open the form, and delegation does not grant them automatically. If the delegate gets an access error on a form, an administrator can assign the appropriate permissions (e.g. through profiles or groups) so the delegate can open the task forms. See xref:runtime:user-delegation.adoc[Delegations].

Reviewer: ```suggestion
Business data (BDM) access is controlled separately from delegation. If a task form reads business data, the delegate must have the same BDM access rights as the delegator to open the form, and delegation does not grant them automatically. If the delegate gets an access error on a form, an administrator can assign the appropriate permissions (e.g. through profiles or groups) so the delegate can open the task forms. See xref:runtime:user-delegation.adoc[Delegations].
```
